src/utils.js

```diff
@@ -64,3 +64,17 @@ export function isEmpty(value) {
   }
   return false;
 }
+
+// We can have initial values undefined, but input makes them empty strings after edit/clear. We will consider those equal.
+
+const withException = (original, current) =>
+  (original === undefined || original === null) && current === ''
+    ? true
+    : undefined;
+
+function isEqualWith(value, other, customizer) {
+  const result = customizer ? customizer(value, other) : undefined;
+  return result === undefined ? value === other : !!result;
+}
+
+export const isEqualByValue = (a, b) => isEqualWith(a, b, withException);
```

Comment:
Perhaps we can simplify all of this to the following?

```js
export function isEqualByValue(original, current) {
  // We can have initial values undefined, but input makes them empty strings
  // after edit/clear. We will consider those equal.
  const isNullish = original == null && current === '';
  return isNullish || original === current;
}
```

Are there any pitfalls I should be aware of with the suggested implementation?

Reply:
I agree, we can simplify it. No need this lodash clone over here :)